test/tests/p5.SoundRecorder.js

```diff
@@ -0,0 +1,142 @@
+'use strict';
+
+define(['chai'], function(chai) {
+  var expect = chai.expect;
+
+  describe('p5.SoundRecorder', function() {
+    var recorder;
+    var inputSoundFile;
+    var originalWriteFile;
+    var writeFileCalled;
+
+    before(function(done) {
+      // clean up audio nodes from other test suites
+      p5.soundOut.soundArray.length = 0;
+
+      // mock p5.prototype.writeFile
+      originalWriteFile = p5.prototype.writeFile;
+      p5.prototype.writeFile = function() {
+        writeFileCalled = true;
+      };
```

Comment:
let's check if p5 is using sinon

Reply:
it looks like p5 does use sinon a bit, but only to stub one specific function:

https://github.com/processing/p5.js/blob/2d4df80e91879c71e34c0fb2553554315d0d4156/test/unit/io/loadImage.js#L6

Reply:
I'm happy to use sinon here if you're ok with it - it's probably better long-term to use a real mocking library